solidity/contracts/hooks/libs/StandardHookMetadata.sol

```diff
@@ -78,6 +79,15 @@ library StandardHookMetadata {
             uint256(bytes32(_metadata[GAS_LIMIT_OFFSET:GAS_LIMIT_OFFSET + 32]));
     }
 
+    function gasLimit(
+        bytes memory _metadata
+    ) internal pure returns (uint256 _gasLimit) {
+        if (_metadata.length < GAS_LIMIT_OFFSET + 32) return 50_000;
+        assembly {
+            _gasLimit := mload(add(_metadata, add(0x20, GAS_LIMIT_OFFSET)))
+        }
+    }
```

Comment:
I dont like the implicit default behavior, and generally we avoid assembly for readability

Reply:
50_000 is taken from `DEFAULT_GAS_USAGE` in the IGP, which is being used as a default gas limit already: https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/46eeda6c3a7110520f4b671a16ed775922d49983/solidity/contracts/hooks/igp/InterchainGasPaymaster.sol#L53

I used assembly because slicing isn't available for memory arrays. We're using `abi.encodePacked` to create the metadata so I can't `abi.decode` to extract the gas limit. I could update `formatMetadata` https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/46eeda6c3a7110520f4b671a16ed775922d49983/solidity/contracts/hooks/libs/StandardHookMetadata.sol#L120 to use a standard ABI encoding, but would need to rework the rest of the functions in that lib to support that.

Reply:
got it
I just dont think its obvious that a `hookMetadata.gasLimit()` defaults to this in the library